core/sql/exp/ExpCriDesc.cpp

```diff
@@ -78,3 +79,20 @@ NA_EIDPROC Lng32 ex_cri_desc::unpack(void * base, void * reallocator)
     }
   return NAVersionedObject::unpack(base, reallocator);
 }
+
+NA_EIDPROC void ex_cri_desc::display(const char* title)
+{
+   cout << title << endl;
+
+   unsigned short tuples = noTuples();
+   char buf[100];
+   for (Int32 i=0; i<tuples; i++) {
+      ExpTupleDesc* tDesc = getTupleDescriptor(i);
+      if ( tDesc ) {
+         sprintf(buf, "%dth tuple:", i);
+         tDesc->display(buf);
+      } else
+         cout << i << "th tuple's TupleDesc is NULL" << endl;
```

Comment:
Will do with next check-in in this area.  Also need to cover more primitive
data types.

On Tue, Feb 16, 2016 at 11:22 AM, Hans Zeller <notifications@github.com>
wrote:

> In core/sql/exp/ExpCriDesc.cpp
> <https://github.com/apache/incubator-trafodion/pull/323#discussion_r53044604>
> :
>
> > @@ -78,3 +79,20 @@ NA_EIDPROC Lng32 ex_cri_desc::unpack(void * base, void * reallocator)
> >      }
> >    return NAVersionedObject::unpack(base, reallocator);
> >  }
> > +
> > +NA_EIDPROC void ex_cri_desc::display(const char* title)
> > +{
> > +   cout << title << endl;
> > +
> > +   unsigned short tuples = noTuples();
> > +   char buf[100];
> > +   for (Int32 i=0; i<tuples; i++) {
> > +      ExpTupleDesc* tDesc = getTupleDescriptor(i);
> > +      if ( tDesc ) {
> > +         sprintf(buf, "%dth tuple:", i);
> > +         tDesc->display(buf);
> > +      } else
> > +         cout << i << "th tuple's TupleDesc is NULL" << endl;
>
> In this case it could display a hex dump of the tupp
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/323/files#r53044604>.
>



-- 
Regards, --Qifan